exercise_solutions.tex

```diff
@@ -1469,58 +1469,213 @@ \subsection*{Solution to \cref{ex:is-conn-trunc-functor}}
-       Let $f:A \to B+C$ be an arbitrary function. We have case analysis on $f(a):B + C$.\\
-       (a) if $f(a) \equiv \inl(u)$ for some $u:B$:
-       We want a function $g:A\to B$ such that $f(x) = \inl(g(x))$ for every $x:A$. Let us define $k:B+C\to B$ by
-       \begin{align*}
-	k(\inl(b)) &\defeq b;\\
-	k(\inr(c)) &\defeq f(a) \enspace,
-       \end{align*}
-       and $g \defeq k \circ f$. Then we have $\inl(g(x)) = \inl(k(f(x))) = f(x)$ for every $x:A$ (by case analysis on $f(x)$ and $\trunc {} {\id[A]{a}{x}}$).
-       With this $g$, we define $h(f) \defeq \inl(g)$. Then we have
-       \begin{displaymath}
-	e_{A,B,C}(h(f)) = e_{A,B,C}(\inl(g)) = \lam{x} \inl(g(x)) = f
-       \end{displaymath}
-       and
-       \begin{displaymath}
-	h(e_{A,B,C}(\inl(v))) = h(\lam{x} \inl(v(x))) = \lam{x} k(\inl(v(x))) = v
-       \end{displaymath}
-       for any $v: A\to B$.\\
-       (b) if $f(a) \equiv \inr(u)$ for some $u:C$: Similar to (a).\\
+ \item Suppose a type $A$ is connected. The goal $\prd{B,C:\type} \isequiv(e_{A,B,C})$ is a mere proposition, so we can assume $a:A$ and $\prd{x,y:A} \brck {x = y}$.
+
+       Let $B,C$ be types.
+       We want to construct $h:(A\to B+C) \to (A\to B)+(A\to C)$ such that
+       $h \circ e_{A,B,C} = \idfunc[(A \to B) + (A \to C)]$ and
+       $e_{A,B,C} \circ h = \idfunc[A \to B + C]$.
+       Let $f:A \to B+C$ be an arbitrary function.
+       Let us define $h(f):(A \to B) + (A \to C)$
+       by case analysis on $f(a):B + C$:
+       \begin{itemize}
+	\item if $f(a) \jdeq \inl(u)$ for some $u:B$:
+	      Let us define $k_1:B+C\to B$ by
+	      \begin{align*}
+	       k_1(\inl(b)) &\defeq b;\\
+	       k_1(\inr(c)) &\defeq u,
+	      \end{align*}
+	      and let $h(f)$ be $\inl(k_1 \circ f)$.
+	\item if $f(a) \jdeq \inr(u)$ for some $u:C$:
+	      Let us define $k_2:B+C\to C$ by
+	      \begin{align*}
+	       k_2(\inl(b)) &\defeq u;\\
+	       k_2(\inr(c)) &\defeq c,
+	      \end{align*}
+	      and let $h(f)$ be $\inr(k_2 \circ f)$.
+       \end{itemize}
+       Let us prove that this $h$ is indeed an inverse of $e_{A,B,C}$.\\
+       First we prove $e_{A,B,C}(h(f)) = f$ for every $f:A \to B + C$
+       by case analysis on $f(a)$:\\
+       \begin{enumerate}[label=(\alph*)]
+        \item if $f(a) \jdeq \inl(u)$ for some $u:B$:
+	      We have $\inl(k_1(f(x))) = f(x)$ for every $x:A$
+	      by case analysis on $f(x)$:
+	      \begin{itemize}
+	       \item if $f(x) \jdeq \inl(v)$ for some $v:B$:
+		     $\inl(k_1(f(x))) = \inl(k_1(\inl(v))) = \inl(v) = f(x)$,
+	       \item if $f(x) \jdeq \inr(v)$ for some $v:C$:
+		     we have $f(a) \neq f(x)$,
+		     which contradicts $\brck {\id[A]{a}{x}}$.
+	      \end{itemize}
+	      Then we have
+	      \begin{displaymath}
+	       e_{A,B,C}(h(f)) = e_{A,B,C}(\inl(k_1 \circ f))
+	       = \lam{x} \inl(k_1(f(x))) = f.
+	      \end{displaymath}
+	      \label{enum:categorical-connectivity-inl}
```

Comment:
If you move this to before the `displaymath`, then it doesn't create extra vertical space.